python/dask_cudf/dask_cudf/accessors.py

```diff
@@ -1,6 +1,62 @@
 # Copyright (c) 2021, NVIDIA CORPORATION.
 
 
+# from cudf.python.dask_cudf.dask_cudf.core import Index
+
+
+class StructMethods:
+    def __init__(self, d_series):
+        self.d_series = d_series
+
+    def field(self, key):
+        """
+        Extract children of the specified struct column
+        in the Series
+        Parameters
+        ----------
+        key: int or str
+            index/position or field name of the respective
+            struct column
+        Returns
+        -------
+        Series
+        Examples
+        --------
+        >>> s = cudf.Series([{'a': 1, 'b': 2}, {'a': 3, 'b': 4}])
+        >>> ds = dask_cudf.from_cudf(s, 2)
+        >>> ds.struct.field(0).compute()
+        0    1
+        1    3
+        dtype: int64
+        >>> ds.struct.field('a').compute()
+        0    1
+        1    3
+        dtype: int64
+        """
+        try:
```

Comment:
Hmm -- this `try...except` logic looks very similar to that in [`cudf/core/column/struct.py:StructMethods`](https://github.com/rapidsai/cudf/blob/59b84f3e8d7cf0d9a15e92bdc8bcafbc01c9bfec/python/cudf/cudf/core/column/struct.py#L162). Can we think of a way to avoid the code duplication between the two places?

Reply:
Perhaps we could add a `valid_struct_field(field)` utility in cudf, and use the same try-except in both places?  The only catch is that the Dask version is just selecting the field from the dtype metadata, because the struct.field call needs to be lazy.

Reply:
Thank you for the comments and suggestions, @shwina & @rjzamora!!  I will go ahead and implement the utility!

Reply:
Sorry @NV-jpt - I was overthinking things in my last suggestion!

You should be able to leverage the existing cudf logic by replacing this entire try/except block with something like:

```python
typ = self.d_series._meta.struct.field(key).dtype
```

This is probably what Ashwin had in mind - A "bad" key should raise one of the errors already included in cudf. So dask-cudf doesn't really need to add a second try/except layer.

Reply:
Thank you, both!!  I have now implemented your suggested changes!  I needed to merge in the changes from PR #[8880 ](https://github.com/rapidsai/cudf/pull/8880) in order for the correct errors to be thrown.  The tests now pass, and everything looks good!